### PR TITLE
Verify Composer installer before executing

### PR DESCRIPTION
**Pull request by thewilkybarkid**
See https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md

### CODE REVIEW

File: elife/composer.sls [hunk 1]
```diff
@@ -75,7 +82,7 @@ update-composer:
         - name: composer self-update
         - onlyif:
             - which composer
-        - require:
+        - require_in:
```
Comment (giorgiosironi) [inline elife/composer.sls]: I guess this is purely for performance reasons?

File: elife/composer.sls [hunk 2]
```diff
@@ -51,14 +51,21 @@ composer-auth:
 {% endif %}
 
 install-composer:
+    file.managed:
+        - name: {{ composer_home }}/setup.php
+        - source: https://getcomposer.org/installer
+        - source_hash: https://composer.github.io/installer.sig
```
Comment (lsh-0) [inline elife/composer.sls]: didn't know this one ...

**Comment by giorgiosironi** [comment]
There are github.com releases but getcomposer.org is the official source.

File: elife/composer.sls (hunk 1 above)
Comment (thewilkybarkid) [inline elife/composer.sls]: Yeah, it installed then immediately checked if it's up to date. (Don't think there's another way to only run the state if other state was skipped.)

**Pull request merged**